auth-plugin.c

```diff
@@ -135,12 +128,7 @@ int mosquitto_auth_acl_check(void *userdata, const char *clientid, const char *u
     return MOSQ_ERR_ACL_DENIED;
   }
 
-  GoString go_clientid = {clientid, strlen(clientid)};
-  GoString go_username = {username, strlen(username)};
-  GoString go_topic = {topic, strlen(topic)};
-  GoInt32 go_access = access;
```

Comment:
This is the one thing that caught my eye.

Why pass `acc` directly and accept a `C.int` on `AuthAclCheck` instead of defining a `GoInt32` as we were doing before?
It's more surprising when you don't give the same treatment to `opts_count`, also defined as a `GoInt32` and passed along to `AuthPluginInit`, which receives a Go `int` instead of a `C.int`.

Reply:
I don't think it really matters technically. I changed this in `AuthAclCheck` for mostly aesthetical reasons, to consistently have C types as parameters. `AuthPluginInit`/`opts_count` is an oversight. Would you prefer to use `C.int` there as well, or should I revert the `acc` parameter of `AuthAclCheck` to `GoInt32`?

Reply:
I don't have a preference really, was just asking if there was a particular reason to treat them differently.
Since there's not, I think we're good to merge.

Thanks!